test/test_indexing.py

```diff
@@ -199,6 +199,18 @@ def test_index_setitem_bools_slices(self):
                 with self.assertRaises(IndexError):
                     a[:] = neg_ones_expanded * 5
 
+    def test_index_scalar_with_bool_mask(self):
+        for device in torch.testing.get_all_device_types():
+            a = torch.tensor(1, device=device)
+            uintMask = torch.tensor(True, dtype=torch.uint8, device=device)
+            boolMask = torch.tensor(True, dtype=torch.bool, device=device)
+            self.assertEqual(a[uintMask], a[boolMask])
+            self.assertEqual(a[uintMask].dtype, a[boolMask].dtype)
```

Comment:
This test feels a little goofy, because aren't you intending to change the `uintMask` behavior in the near future?

Reply:
Well once its changed i will update the test but for now i want to show that the result is identical, as expected.